cardano-api/src/Cardano/Api/KeysByron.hs

```diff
@@ -144,3 +175,119 @@ instance CastVerificationKeyRole ByronKey PaymentKey where
       . (castVerificationKey :: VerificationKey ByronKey
                              -> VerificationKey PaymentExtendedKey)
 
+instance IsByronKey ByronKey where
+  byronKeyFormat = ByronNonLegacyKeyFormat
+
+--
+-- Legacy Byron key
+--
+
+instance Key ByronKeyLegacy where
+
+    newtype VerificationKey ByronKeyLegacy =
+           ByronVerificationKeyLegacy Byron.VerificationKey
+      deriving stock (Eq)
+      deriving (Show, IsString) via UsingRawBytesHex (VerificationKey ByronKeyLegacy)
+      deriving newtype (ToCBOR, FromCBOR)
+      deriving anyclass SerialiseAsCBOR
+
+    newtype SigningKey ByronKeyLegacy =
+           ByronSigningKeyLegacy Byron.SigningKey
+      deriving (Show, IsString) via UsingRawBytesHex (SigningKey ByronKeyLegacy)
+      deriving newtype (ToCBOR, FromCBOR)
+      deriving anyclass SerialiseAsCBOR
+
+    deterministicSigningKey :: AsType ByronKeyLegacy -> Crypto.Seed -> SigningKey ByronKeyLegacy
+    deterministicSigningKey _ _ = error "Please generate a non legacy Byron key instead"
+
+    deterministicSigningKeySeedSize :: AsType ByronKeyLegacy -> Word
+    deterministicSigningKeySeedSize AsByronKeyLegacy = 32
+
+    getVerificationKey :: SigningKey ByronKeyLegacy -> VerificationKey ByronKeyLegacy
+    getVerificationKey (ByronSigningKeyLegacy sk) =
+      ByronVerificationKeyLegacy (Byron.toVerification sk)
+
+    verificationKeyHash :: VerificationKey ByronKeyLegacy -> Hash ByronKeyLegacy
+    verificationKeyHash (ByronVerificationKeyLegacy vkey) =
+      ByronKeyHashLegacy (Byron.hashKey vkey)
+
+instance HasTypeProxy ByronKeyLegacy where
+  data AsType ByronKeyLegacy = AsByronKeyLegacy
+  proxyToAsType _ = AsByronKeyLegacy
+
+instance HasTextEnvelope (VerificationKey ByronKeyLegacy) where
+    textEnvelopeType _ = "PaymentVerificationKeyByronLegacy_ed25519_bip32"
+
+instance HasTextEnvelope (SigningKey ByronKeyLegacy) where
+    textEnvelopeType _ = "PaymentSigningKeyByronLegacy_ed25519_bip32"
```

Comment:
Do we need text envelope instances for them? Don't we just read them in in the old formats and convert to the new? So we'd never be writing out text envelope wrapper with the legacy key format embedded? So if it's not used, perhaps best to remove this instance to help people avoid that mistake. And if it is used, why? :smiley:

Reply:
Copy pasted without thinking!

Reply:
Ok. Fine.

Maybe we ought to consider relaxing that superclass constraint, but it's fine for now.